test/version.js

```diff
@@ -1,139 +1,229 @@
+	t.is(new Version('0.0.0').setFrom('1.0.0-0').format(), newVersion);
 });
 
-test('version.isGreaterThanOrEqualTo', t => {
-	t.false(new Version('1.0.0').isGreaterThanOrEqualTo('0.0.1'));
-	t.false(new Version('1.0.0').isGreaterThanOrEqualTo('0.1.0'));
+test('format - preminor', t => {
+	const newVersion = makeNewFormattedVersion('0.{1}.0-{0}');
 
-	t.false(new Version('1.0.0').isGreaterThanOrEqualTo('1.0.0-0'));
-	t.false(new Version('1.0.0').isGreaterThanOrEqualTo('1.0.0-beta'));
+	t.is(new Version('0.0.0').setFrom('preminor').format(), newVersion);
+	t.is(new Version('0.0.0').setFrom('0.1.0-0').format(), newVersion);
+});
 
-	t.true(new Version('1.0.0').isGreaterThanOrEqualTo('1.0.0'));
-	t.true(new Version('1.0.0').isGreaterThanOrEqualTo('1.0.1'));
-	t.true(new Version('1.0.0').isGreaterThanOrEqualTo('1.1.0'));
-	t.true(new Version('1.0.0').isGreaterThanOrEqualTo('2.0.0'));
+test('format - prepatch', t => {
+	const newVersion = makeNewFormattedVersion('0.0.{1}-{0}');
 
-	t.true(new Version('1.0.0').isGreaterThanOrEqualTo('2.0.0-0'));
-	t.true(new Version('1.0.0').isGreaterThanOrEqualTo('2.0.0-beta'));
+	t.is(new Version('0.0.0').setFrom('prepatch').format(), newVersion);
+	t.is(new Version('0.0.0').setFrom('0.0.1-0').format(), newVersion);
 });
 
-test('version.isLowerThanOrEqualTo', t => {
-	t.true(new Version('1.0.0').isLowerThanOrEqualTo('0.0.1'));
-	t.true(new Version('1.0.0').isLowerThanOrEqualTo('0.1.0'));
+test('format - prerelease', t => {
+	const newVersion = makeNewFormattedVersion('0.0.0-{1}');
 
-	t.true(new Version('1.0.0').isLowerThanOrEqualTo('1.0.0-0'));
-	t.true(new Version('1.0.0').isLowerThanOrEqualTo('1.0.0-beta'));
-	t.true(new Version('1.0.0').isLowerThanOrEqualTo('1.0.0'));
+	t.is(new Version('0.0.0-0').setFrom('prerelease').format(), newVersion);
+	t.is(new Version('0.0.0-0').setFrom('0.0.0-1').format(), newVersion);
+});
 
-	t.false(new Version('1.0.0').isLowerThanOrEqualTo('1.0.1'));
-	t.false(new Version('1.0.0').isLowerThanOrEqualTo('1.1.0'));
-	t.false(new Version('1.0.0').isLowerThanOrEqualTo('2.0.0'));
+test('format - prerelease as prepatch', t => {
+	const newVersion = makeNewFormattedVersion('0.0.{1}-{0}');
 
-	t.false(new Version('1.0.0').isLowerThanOrEqualTo('2.0.0-0'));
-	t.false(new Version('1.0.0').isLowerThanOrEqualTo('2.0.0-beta'));
+	t.is(new Version('0.0.0').setFrom('prerelease').format(), newVersion);
+	t.is(new Version('0.0.0').setFrom('0.0.1-0').format(), newVersion);
 });
 
-test('version.satisfies', t => {
+test('format - prerelease with multiple numbers', t => {
+	const newVersion = makeNewFormattedVersion('0.0.{1}-{0.0}'); // TODO: should it be {0}.{0}?
+	t.is(new Version('0.0.0').setFrom('0.0.1-0.0').format(), newVersion);
+});
+
+test('format - prerelease with text', t => {
+	const newVersion = makeNewFormattedVersion('0.0.{1}-{alpha.0}'); // TODO: should it be {alpha}.{0}?
+	t.is(new Version('0.0.0').setFrom('0.0.1-alpha.0').format(), newVersion);
+});
```

Comment:
Should prerelease formatting highlight the whole prerelease part, or just each individual part (i.e. exlcuding `.`s)?

![image](https://github.com/sindresorhus/np/assets/1403701/a7b3638e-2e72-4a21-939f-4fa8fc799ad0)

Reply:
The first one (all).

Reply:
Great, that's the current behavior.